qupath-gui-fx/src/main/java/qupath/lib/gui/commands/PathObjectGridView.java

```diff
@@ -213,6 +213,7 @@ private static List<PathObject> getAnnotations(PathObjectHierarchy hierarchy) {
 	public Stage getStage() {
 		if (stage == null)
 			initializeGUI();
+			stage.setWidth(850);
```

Comment:
Can be set smaller for sure, the reason for 850 was to include all the menu bar options and during testing I always needed access to the "size" in the expandable area.
Here is the comparison 
![image](https://github.com/qupath/qupath/assets/42358257/eb62db9b-0bcd-449d-985d-06fc5350fe4c)
![image](https://github.com/qupath/qupath/assets/42358257/4a25159f-0db4-4673-8bc6-8aae1e642f52)
Certainly in support of a solution that determines the largest QuPath window size and sets accordingly from that

Reply:
Seems like we should really make a thing Stage subclass that handles different resolutions elegantly, because while I agree we should avoid making anything too big for a screen, it's also a bit silly to make every window tiny when a lot of folks are rocking at least 1440p displays at this point.

Reply:
Changed it back to 600 for now

Reply:
I imagine in qupath-fxtras we could have a method something like
```java
public static Dimension calculateScreenSize(Window parent, double preferredWidth, double preferredHeight)
```
which should return the preferred dimensions if they fit, and smaller ones if they don't. The `parent` would be needed to find which screen to use, in case there are multiple.

That feels a bit safer to me that subclassing `Stage`.

But yeah, fair point that needing to resize every time is also annoying. I guess any reasonable size is ok for now, and we can revisit if we get a general size-choosing solution. I'd vote to go annoyingly small rather than risk too big, but I don't have a very strong opinion on it if we check the behavior is tolerable if the width is larger than the screen width.

(If there's any way to coax JavaFX into determining a sensible default that fits the toolbar without going beyond the screen, that'd be even better - but I'm not sure if there is.)

Reply:
Made a PR at https://github.com/qupath/qupath/pull/1480
Was going to add the method to qupath-fxtras, but in the end decided it wasn't that much code required to add it on a case-by-case basis, once we get the screen containing the parent with `FXUtils`.